### PR TITLE
Make residuals_plot quick method primetime

### PR DESCRIPTION
**Pull request by bbengfort**
Refactors the residuals_plot quick method to the new quick method signature and verifies that it works correctly. Tasks:

- [x] check quick method docstring and signature
- [x] implement the new train/test split mechanism
- [x] add the show kwarg and implementation
- [x] verify the quick method tests
- [x] run locally to ensure it works as expected
- [x] update the documentation

Related to #960

### CODE REVIEW

File: tests/test_regressor/test_residuals.py [hunk 1]
```diff
@@ -343,11 +343,50 @@ def test_residuals_quick_method(self):
...
+        """
+        msg = "both X_test and y_test are required if one is specified"
+        with pytest.raises(YellowbrickValueError, match=msg):
+            residuals_plot(
+                Lasso(),
+                self.data.X.train,
+                self.data.y.train,
+                self.data.X.test,
+            )
+
```
Comment (rebeccabilbro) [inline tests/test_regressor/test_residuals.py]: Thank you for adding these tests to ensure the edge cases of a user passing in only training data (which should work as normal), and a user passing in incomplete training and test splits (which should not)!

File: yellowbrick/regressor/residuals.py [hunk 2]
```diff
@@ -790,15 +799,24 @@ def residuals_plot(
...
-    visualizer.finalize()
+    # Score the visualizer
+    if X_test is not None and y_test is not None:
+        viz.score(X_test, y_test)
+    elif X_test is not None or y_test is not None:
+        raise YellowbrickValueError(
+            "both X_test and y_test are required if one is specified"
+        )
+    else:
+        viz.score(X_train, y_train)
```
Comment (rebeccabilbro) [inline yellowbrick/regressor/residuals.py]: Thank you for adding in this logic to protect against edge cases &mdash; this might be a useful snippet for others to use as a reference for other `ScoreVisualizer` quick methods!
